sdk/nodejs/cloudwatch/metric.ts

```diff
@@ -0,0 +1,402 @@
+    /**
+     * The namespace for this metric. See docs for the
+     * [list of namespaces](https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/aws-namespaces.html).
+     * See docs for
+     * [supported metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/DeveloperGuide/CW_Support_For_AWS.html).
+     */
+    public readonly namespace: pulumi.Output<string>;
+    /**
+     * The period in seconds over which the specified [statistic] is applied.  Must be in multiples
+     * of 60. Periods are defined in numbers of seconds, and valid values for period are 1, 5, 10,
+     * 30, or any multiple of 60. For example, to specify a period of six minutes, use 360 as the
+     * period value. You can adjust how the data is aggregated by varying the length of the period.
+     * A period can be as short as one second or as long as one day (86,400 seconds). The default
+     * value is 60 seconds.
+     *
+     * See https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html
+     * for more details.
+     */
+    public readonly period: pulumi.Output<number>;
+    /**
+     * The statistic to apply to the alarm's associated metric. Either of the following is
+     * supported: `SampleCount`, `Average`, `Sum`, `Minimum`, `Maximum`.
+     *
+     * Defaults to [Average] if [statistic] and [extendedStatistic] is unspecified.
+     */
+    public readonly statistic: pulumi.Output<MetricStatistic>;
+    /**
+     * The percentile statistic for the metric associated with the alarm. Specify a value between
+     * [0.0] and [100].
+     */
+    public readonly extendedStatistic: pulumi.Output<number | undefined>;
+    /**
+     * The unit for this metric.
+     *
+     * See https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDatum.html
+     */
+    public readonly unit: pulumi.Output<MetricUnit | undefined>;
+
+    /**
+     * @param resource Optional resource this is a metric for.  This is only used for parenting
+     * purposes.  i.e. if an [Alarm] is created from this [Metric], then [resource] will be used as
+     * the parent of the alarm by default.
+     */
+    constructor(args: MetricArgs, resource?: pulumi.Resource) {
+        this.resource = resource;
+        this.name = pulumi.output(args.name);
+        this.dimensions = pulumi.output(args.dimensions);
+        this.namespace = pulumi.output(args.namespace);
+        this.period = utils.ifUndefined(args.period, 60).apply(validatePeriod);
+        this.statistic = pulumi.all([args.statistic, args.extendedStatistic])
+                               .apply(([statistic, extendedStatistic]) => validateStatistics(statistic, extendedStatistic));
+        this.extendedStatistic = pulumi.output(args.extendedStatistic).apply(validateExtendedStatistic);
+        this.unit = pulumi.output(args.unit);
+    }
+
+    public with(change: MetricChange | undefined) {
```

Comment:
helper method so that one can easily take the default metrics exposed by packages, and tweak what representation you want.  Common things to do would be to change the unit, period or statistic you care about hte metric for.  for example, while the default period for metrics is 60 seconds, you could always call `metricXXX({ period: 120 })` to ask for the same metric for a 120 second period.